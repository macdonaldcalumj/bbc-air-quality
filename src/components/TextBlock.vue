<template>
  <section class="section">
    <div class="content">
      <p v-for="p in paragraphs" :key="p">
        {{ p }}
      </p>
    </div>
  </section>
</template>

<script>
export default {
  props: {
    paragraphs: {
      default() {
        return []
      }
    },
  }
}
</script>
