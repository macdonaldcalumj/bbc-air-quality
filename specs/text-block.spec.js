import { mount } from "@vue/test-utils";
import TextBlock from '@/components/TextBlock'

test("test mounting component", () => {
  const wrapper = mount(TextBlock)

  expect(wrapper).toMatchSnapshot()
})

test("test props", async () => {
  const wrapper = mount(TextBlock,
    {
      propsData: {
        paragraphs: ['a']
      }
    }
  )

  expect(wrapper).toMatchSnapshot();
  
  await wrapper.setProps({ paragraphs: ['a', 'b'] })

  expect(wrapper).toMatchSnapshot()
});
