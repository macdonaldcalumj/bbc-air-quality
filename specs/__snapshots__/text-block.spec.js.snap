// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test mounting component 1`] = `
<section class="section">
  <div class="content"></div>
</section>
`;

exports[`test props 1`] = `
<section class="section">
  <div class="content">
    <p>
      a
    </p>
  </div>
</section>
`;

exports[`test props 2`] = `
<section class="section">
  <div class="content">
    <p>
      a
    </p>
    <p>
      b
    </p>
  </div>
</section>
`;
